v2/tools/generator/internal/config/literal_matcher_test.go

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestLiteralMatcher_Matches_GivesExpectedResults(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name     string
+		literal  string
+		value    string
+		expected bool
+	}{
+		{"Exact match", "Foo", "Foo", true},
+		{"Uppercase literal matches lowercase value", "FOO", "foo", true},
+		{"Lowercase literal matches uppercase value", "foo", "FOO", true},
+		{"Mixed case literal matches different mixed case value", "Foo", "foO", true},
+		{"Different strings do not match", "Foo", "Bar", false},
+		{"Leading whitespace does not match", "  Foo", "Foo", false},
+		{"Trailing whitespace does not match", "Foo  ", "Foo", false},
```

Comment:
These are a bit surprising - wouldn't it be nicer to handle these cases and succeed? Or is there some reason a literal matcher shouldn't match them?

Reply:
I don't really have much of an opinion, included the tests as documentation. Changed to be whitespace tolerant.